stairway/src/main/java/bio/terra/stairway/FlightMap.java

```diff
@@ -55,130 +46,114 @@ public FlightMap() {
    */
   List<FlightInput> makeFlightInputList() {
     ArrayList<FlightInput> inputList = new ArrayList<>();
-    for (Map.Entry<String, Object> entry : map.entrySet()) {
-      try {
-        String value = getObjectMapper().writeValueAsString(entry.getValue());
-        inputList.add(new FlightInput(entry.getKey(), value));
-      } catch (JsonProcessingException ex) {
-        throw new JsonConversionException("Failed to convert value to json string", ex);
-      }
+    for (Map.Entry<String, String> entry : map.entrySet()) {
+      inputList.add(new FlightInput(entry.getKey(), entry.getValue()));
     }
     return inputList;
   }
 
   /**
    * Depending on what version of code a Flight log was written with, we may have a JSON Map, a
    * {@code List<FlightInput>}, both, or neither at deserialization time. This method is used to
-   * generate an {@code Optional<FlightMap>} based on what was contained in the database. This will
-   * return an empty {@code Optional<FlightMap>} if json is null.
+   * generate a FlightMap based on what was contained in the database.
```

Comment:
I'd like an explanation on which version maps to which conditions and especially what the current state is.
Is there a follow up ticket to eventually remove support for the old version, or does it need to be supported indefinitely?

Reply:
I updated the PR description to link out to [the document](https://docs.google.com/document/d/1b1VckTBb_V1pCMhoN2GgHnGDR2f8MV2HuwIgDqFc_9M/edit) describing the phases of this change.  

Current services are all in Phase 1 (last to go was TDR back in mid-May), where we are writing both the working/output parameter fields as JSON maps and writing to the new `flightworking` table.  We are also doing informational checks to make sure that the working/output maps and `flightworking` contents match up (where `flightworking` rows are present).

These checks are informational only and log ERRORs (but do not result in failures).  There was actually a bug in the verification logic that showed some false positives (PF-709) which @ddietterich observed in WSM testing, which I think gives us some confidence that we are not seeing mismatches here.

I created a ticket [PF-917] for removal of these columns.  This was initially slated as "not POR" for this process due to the number of old flights hanging around (esp in TDR), but there may be an opportunity to do this sooner now that we have implemented grooming of old flights.

Reply:
IIUC, if all Stairway clients have moved to Phase 1, then there is no timing issue with installing Phase 2.

When we do PF-917, then clients need to ensure that there are no pre-phase 2 flights in their databases. We can add a `stairctl` command to do that check.

Reply:
Phase 2 is not backward compatible, so there's no rolling back from this (which is why we were conservative and did the schema update separately then let this run a while with the ERROR logging in place).

Am I correct in my assumption that absent a rollback, we do not have any scenarios where we will try to read a flight written with Phase 2 code using Phase 1 code?

Reply:
That is my understanding also.